confmap/provider.go

```diff
@@ -6,11 +6,23 @@ package confmap // import "go.opentelemetry.io/collector/confmap"
 import (
 	"context"
 	"fmt"
+
+	"go.uber.org/zap"
 )
 
 // ProviderSettings are the settings to initialize a Provider.
 // Any Provider should take this as a parameter in its constructor.
-type ProviderSettings struct{}
+type ProviderSettings struct {
+	Logger *zap.Logger
+}
+
+func NewProviderSettings(logger *zap.Logger) ProviderSettings {
+	return ProviderSettings{Logger: logger}
+}
+
+func NewProviderSettingsNoopLogger() ProviderSettings {
+	return NewProviderSettings(zap.NewNop())
+}
```

Comment:
When we actually have a useful logger, we would likely want to change this. To avoid having to go through breaking changes, I would define this on an `internal` package. We can either define this in a package within `confmap/internal` or, since we use these functions in `otelcol`, just on the top level internal folder

Reply:
This is kinda tricky actually...
If i move just `NewProviderSettings` to confmap/internal then `NewProviderSettingsNoopLogger` has to call NewProviderSettings which imports ProviderSettings which is an import loop.  If I change `NewProviderSettingsNoopLogger` to just initialize the struct itself it defeats the point of having `NewProviderSettings` at all

If I move both functions in there, `NewProviderSettingsNoopLogger` can't be accessed everywhere it's needed.

I'm not sure about the utility about moving these into the top level internal folder either?  If its in the top level folder can't anyone access the functions?  Apologies if this is obvious honestly my Go package knowledge is kinda rusty.

At this point I'm tempted to just get rid of `NewProviderSettings` entirely...

Reply:
>  If its in the top level folder can't anyone access the functions?

`<prefix>/internal/<suffix>` can only be imported in packages whose path shares `<prefix>`. So moving to a top-level internal folder (`go.opentelemetry.io/collector/internal/<something>`) means that anything within this repository can access it, but nothing outside of it can.

> At this point I'm tempted to just get rid of NewProviderSettings entirely...

I am also fine with that, it leads to some repetition but it should be temporary